### PR TITLE
[WIP] Add cert+key hash to genevalogging deployment doc

### PR DESCRIPTION
**Pull request by petrkotas**
Introduce hash to genevalogging deployment.
It enables auto redeploy of logging daemonset
when the cert+key secret changes.

/cc @mjudeikis 

Signed-off-by: Petr Kotas <pkotas@redhat.com>

### CODE REVIEW

File: pkg/genevalogging/genevalogging.go [hunk 1]
```diff
@@ -362,9 +372,15 @@ func (g *genevaLogging) CreateOrUpdate(ctx context.Context) error {
...
+			UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+				Type: appsv1.RollingUpdateDaemonSetStrategyType,
+				RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+					MaxUnavailable: &maxUnavailablePods,
+				},
+			},
 			Template: v1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
-					Labels:      map[string]string{"app": "mdsd"},
+					Labels:      map[string]string{"app": "mdsd", "certHash": certHash},
```
Comment (asalkeld) [inline pkg/genevalogging/genevalogging.go:383]: i do this in the operator, just differently: https://github.com/Azure/ARO-RP/blob/aro-operator/pkg/controllers/genevalogging_controller.go#L109
https://github.com/Azure/ARO-RP/blob/aro-operator/pkg/dynamichelper/workloadhelper.go

File: pkg/genevalogging/genevalogging.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ package genevalogging
 
 import (
 	"context"
+	"crypto/sha1"
```
Comment (jim-minter) [inline pkg/genevalogging/genevalogging.go:8]: we're audited on our choice of cryptographic functions.  It makes no odds here but use sha256 please

**Comment by petrkotas** [comment]
Thanks, @asalkeld for pointing me to `aro-operator`. After auditing, I agree that it is the better place for that code.

I am still missing the enablement for rolling updates. I left it in this PR.

This can be added later on.

**Comment by mjudeikis** [comment]
Lets re-open when needed

**Pull request closed without merging**